clients/node/client-rds-data-node/protocol/AwsRestJson1_1.ts

```diff
@@ -0,0 +1,297 @@
+import {
+  ColumnMetadata,
+  ExecuteStatementRequest,
+  ExecuteStatementResponse,
+  Field,
+  SqlParameter
+} from "../models/rdsdataservice";
+import { HttpRequest, HttpResponse } from "@aws-sdk/protocol-http";
+import * as __aws_sdk_stream_collector_node from "@aws-sdk/stream-collector-node";
+import * as __aws_sdk_util_utf8_node from "@aws-sdk/util-utf8-node";
+
+export function executeStatementAwsRestJson1_1Serialize(
+  input: ExecuteStatementRequest
+): HttpRequest {
+  let body: any = {};
+  if (input.resourceArn !== undefined) {
+    body.resourceArn = input.resourceArn;
+  }
+
+  if (input.secretArn !== undefined) {
+    body.secretArn = input.secretArn;
+  }
+
+  if (input.sql !== undefined) {
+    body.sql = input.sql;
+  }
+
+  if (input.database !== undefined) {
+    body.database = input.database;
+  }
+
+  if (input.schema !== undefined) {
+    body.schema = input.schema;
+  }
+
+  if (input.parameters !== undefined) {
+    body.parameters = sqlParameterListAwsRestJson1_1Serialize(input.parameters);
+  }
+
+  if (input.transactionId !== undefined) {
+    body.transactionId = input.transactionId;
+  }
+
+  if (input.includeResultMetadata !== undefined) {
+    body.includeResultMetadata = input.includeResultMetadata;
+  }
+
+  if (input.continueAfterTimeout !== undefined) {
+    body.continueAfterTimeout = input.continueAfterTimeout;
+  }
+
+  return new HttpRequest({
+    body: JSON.stringify(body),
+    path: "/execute",
+    method: "POST",
+    protocol: "https:",
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+}
+
+export function executeStatementAwsRestJson1_1Deserialize(
+  output: HttpResponse
+): Promise<ExecuteStatementResponse> {
+  if (output.statusCode !== 200) {
+    return executeStatementAwsRestJson1_1DeserializeError(output);
+  }
+  let data: any = parseBody(output.body);
+  return Promise.resolve({
+    __type: "com.amazon.rdsdataservice#ExecuteStatementResponse",
+    records: recordsAwsRestJson1_1Deserialize(data.records),
+    columnMetadata: columnMetadataListAwsRestJson1_1Deserialize(
+      data.columnMetadata
+    ),
+    numberOfRecordsUpdated: data.numberOfRecordsUpdated,
+    generatedFields: generatedFieldsAwsRestJson1_1Deserialize(
+      data.generatedFields
+    )
+  });
+}
+
+function executeStatementAwsRestJson1_1DeserializeError(
```

Comment:
I think this needs work. It should check the x-amzn-ErrorType header for the ID of the error and use that. If that's not found, then it should fall back to status codes.